jenkins/check_projects.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Check the iam rights of test projects, optionally fixing them
+
+Usage:
+  python update_jobs.py <config|--default> [--fix]
+
+Config format: {<ROLE>: [MEMBER, ...]}
+  {
+    "roles/editor": [
+      "serviceAccount:foo@bar.com",
+      "user:me@you.com",
+      "group:whatever@googlegroups.com"],
+    "roles/viewer": [...],
+    ...
+  }
+"""
+
+
+import argparse
+import collections
+import json
+import os.path
+import re
+import subprocess
+import sys
+import threading
+
+semaphore = threading.Semaphore(10)  # 10 concurrent gcloud calls
+lock = threading.Lock()  # Don't print/append concurrently
+null = open('/dev/null', 'w')
+
+completed = set()  # Projects with no updated needed
+updated = set()  # Projects which changed
+errors = set()  # Projects which failed to fully update
+helpers = {}  # People with owners rights to update error projects
```

Comment:
Did some brief googling, and it seems dictionary access from multiple threads is "safe" in the sense that it won't corrupt entries, because the GIL will serialize calls in some undefined way. So concurrent writes to `helpers` will result in one of them winning, and the fact that you put each project `Check` call in a thread means this should be fine, it just looks nasty. If you put a comment here that explains why it's ok to write to these from separate threads I won't block you.